lib/game.rb

```diff
@@ -78,5 +94,14 @@ def output_win
     def output_tie
         game_io.output_message(Messages.tie) 
     end
+
+    def save_game
+        repository_type = game_io.is_a?(ConsoleIO) ? Memory.new : Local.new
```

Comment:
Why do we need to inspect the class of `game_io` to determine which type of repository to use?

Reply:
Didn't think this through enough -- adjusted to use dependency injection into Game itself, with a default of Local.new